File: 257-binary-tree-paths/binary-tree-paths.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
     public List<String> binaryTreePaths(TreeNode root) {
       List<String> arr= new ArrayList<>();
       if(root==null) return arr;
       String st = String.valueOf(root.val);
       path(root,st,arr);
       return arr;
    }
    public static void path(TreeNode root, String st, List<String> arr){
        if(root.left==null && root.right==null){
            arr.add(st);
            return;
        }
        //st+=String.valueOf(root.val);
        if(root.left!=null){
            path(root.left,st+"->"+String.valueOf(root.left.val),arr);
        }
        if(root.right!=null){
            path(root.right,st+"->"+String.valueOf(root.right.val),arr);
        }
    }
}

File: 257-binary-tree-paths/binary-tree-paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<string> binaryTreePaths(TreeNode* root) {
        
        vector<string> arr;
        // string

        if(root==nullptr) return arr;
       
        fun(root,arr,"");
       
        return arr;
    }
    public:
        void fun(TreeNode* root, vector<string> &arr,string path){
            if(root==nullptr) return ;
           
            
            if(path.empty()){
                path = to_string(root->val);
            }
            else{
                path += "->" + to_string(root->val);
            }

             if(root->left==nullptr && root->right==nullptr){
                  arr.push_back(path);
            return;
            }

            fun(root->left,arr,path);
            fun(root->right,arr,path);
        }
};